src/main/java/cs2103/v15_1j/jimjim/storage/JJStorage.java

```diff
@@ -51,94 +39,56 @@ public JJStorage() {
-        return savedDeadlineTasksFile;
-    }
-
-    public File getSavedEventsFile() {
-        return savedEventsFile;
+    public File getSaveFile() {
+        return saveFile;
     }
 
     @Override
     public DataLists load() {
-        BufferedReader tasksBufferedReader;
-        BufferedReader deadlineTasksBufferedReader;
-        BufferedReader eventsBufferedReader;
+        BufferedReader reader = null;
         try {
             // Reads data from file
-            tasksBufferedReader = new BufferedReader(new FileReader(savedFloatingTasksFile));
-            deadlineTasksBufferedReader = new BufferedReader(new FileReader(savedDeadlineTasksFile));
-            eventsBufferedReader = new BufferedReader(new FileReader(savedEventsFile));
+            reader = new BufferedReader(new FileReader(saveFile));
+            // Converts read data back into Java types
+            return gson.fromJson(reader, dataListsType);
         } catch (FileNotFoundException e) {
             return new DataLists();
+        } finally {
+            try {
+                if (reader != null) {
+                    reader.close();
+                }
+            } catch (IOException e) {
+                // TODO Auto-generated catch block
+                e.printStackTrace();
+            }
         }
-        // Converts read data back into Java types
-        List<FloatingTask> floatingTasksList = gson.fromJson(tasksBufferedReader, listOfFloatingTaskType);
-        List<DeadlineTask> deadlineTasksList = gson.fromJson(deadlineTasksBufferedReader, listOfDeadlineTaskType);
-        List<Event> eventsList = gson.fromJson(eventsBufferedReader, listOfEventType);
-
-        try {
-            tasksBufferedReader.close();
-            deadlineTasksBufferedReader.close();
-            eventsBufferedReader.close();
-        } catch (IOException e) {
-            // TODO Auto-generated catch block
-            e.printStackTrace();
-        }
-
-        return new DataLists(deadlineTasksList, floatingTasksList, eventsList);
     }
 
     @Override
     public boolean save(DataLists list) {
-        List<FloatingTask> tasksList = list.getFloatingTasksList();
-        List<DeadlineTask> deadlineTasksList = list.getDeadlineTasksList();
-        List<Event> eventsList = list.getEventsList();
-
-        // Convert each list to separate JSON string
-        String tasksJSON = gson.toJson(tasksList, listOfFloatingTaskType);
-        String deadlineTasksJSON = gson.toJson(deadlineTasksList, listOfDeadlineTaskType);
-        String eventsJSON = gson.toJson(eventsList, listOfEventType);
+        String dataListsJSON = gson.toJson(list, dataListsType);
 
-        return writeJSONToFile(tasksJSON, deadlineTasksJSON, eventsJSON);
+        return writeJSONToFile(dataListsJSON, saveFile);
     }
 
-    private boolean writeJSONToFile(String tasksJSON, String deadlineTasksJSON, String eventsJSON) {
+    private boolean writeJSONToFile(String jsonString, File file) {
```

Comment:
note that I am adding the `File` argument instead of just using `saveFile` so that later when we save configurations we can use this function